components/dashboard/src/admin/TeamDetail.tsx

```diff
@@ -151,6 +225,89 @@ export default function TeamDetail(props: { team: Team }) {
                     ))
                 )}
             </ItemsList>
+            <Modal
+                visible={editSpendingLimit}
+                onClose={() => setEditSpendingLimit(false)}
+                title="Update Usage Limit"
+                onEnter={() => false}
+                buttons={[
+                    <button
+                        className="secondary"
+                        disabled={usageLimit === costCenter?.spendingLimit}
+                        onClick={async () => {
+                            if (usageLimit !== undefined) {
+                                await getGitpodService().server.adminSetUsageLimit(
+                                    AttributionId.render(AttributionId.create(team)),
+                                    usageLimit || 0,
+                                );
+                                setUsageLimit(undefined);
+                                initialize();
+                                setEditSpendingLimit(false);
+                            }
+                        }}
+                    >
+                        Save
+                    </button>,
+                ]}
+            >
+                <p>Update the spending limit for this account.</p>
+                <div className="flex flex-col">
+                    <input
+                        type="number"
+                        min={usage?.creditsUsed}
+                        max={500000}
+                        title="Update Usage Limit"
+                        value={usageLimit}
+                        onChange={(event) => setUsageLimit(Number.parseInt(event.target.value))}
+                    />
+                </div>
+            </Modal>
+            <Modal
+                onEnter={() => false}
+                visible={editAddCreditNote}
+                onClose={() => setEditAddCreditNote(false)}
+                title="Add Credits"
+                buttons={[
+                    <button
+                        className="secondary"
+                        disabled={creditNote.credits === 0 || !creditNote.note}
+                        onClick={async () => {
+                            if (creditNote.credits !== 0 && !!creditNote.note) {
+                                await getGitpodService().server.adminAddUsageCreditNote(
+                                    AttributionId.render(AttributionId.create(team)),
+                                    creditNote.credits,
+                                    creditNote.note,
+                                );
+                                setEditAddCreditNote(false);
+                                setCreditNote({ credits: 0 });
+                                initialize();
+                            }
+                        }}
+                    >
+                        Save
+                    </button>,
+                ]}
+            >
+                <p>Adds or subtracts the amount of credits from this account.</p>
```

Comment:
**thought**: An info alert at the top of this modal to explain what's happening or when and how to use this could help, as we do with [`/admin/blocked-repositories`](https://gitpod.io/admin/blocked-repositories). But we can add this later if needed.

| Blocked Repositories Modal | Usage Limit Modal |
|-|-|
| <img width="540" alt="Screenshot 2022-11-21 at 2 30 52 PM" src="https://user-images.githubusercontent.com/120486/203056070-5bcda3e7-845e-4c16-91a5-89b49b22214f.png"> | <img width="545" alt="Screenshot 2022-11-21 at 2 31 17 PM" src="https://user-images.githubusercontent.com/120486/203056176-6ddc98d8-4c78-4f26-b4d4-3b29203b0f62.png"> |